utilities/xmlrpcpp/src/XmlRpcDispatch.cpp

```diff
@@ -176,12 +176,16 @@ XmlRpcDispatch::work(double timeout)
     {
 #if defined(_WINDOWS)
       XmlRpcUtil::error("Error in XmlRpcDispatch::work: error in poll (%d).", WSAGetLastError());
+      _inWork = false;
+      return;
 #else
       if(errno != EINTR)
+      {
         XmlRpcUtil::error("Error in XmlRpcDispatch::work: error in poll (%d).", nEvents);
+        _inWork = false;
```

Comment:
Also, `timeout_ms` has to be recomputed from `_end_time` and `getTime()`. There is probably a larger problem here with `timeout_ms`, as the current implementation calls all `poll()`s with `timeout_ms` regardless of how much time has elapsed from the beginning of the call. Maybe the recomputation of `timeout_ms` should be moved right before the `poll()` call.